### PR TITLE
Custom rtapp v3

### PR DESCRIPTION
**Pull request by credp**
Hi, I wanted to do a bit more synthetic modelling and the built-in generators weren't quite what I needed so I thought about using the custom generator. I had a couple of problems getting it to work, and in the process I added the ability to use either an existing file or a variable.

There's an example too.

Hopefully the indentation is preferable :)

### CODE REVIEW

File: libs/wlgen/wlgen/rta.py [hunk 1]
```diff
@@ -266,19 +266,29 @@ def _confCustom(self):
...
+	# check for inline config
+	if 'custom_config' in self.params['custom']:
+        # inline config present, turn it into a file repr so we can parse the same
+        tmp_json = json.dumps(self.params['custom']['custom_config'],
+            indent=4, separators=(',', ': '), sort_keys=True)
+        ifile = tmp_json.split('\n')
+	else:
+        # open the supplied filename
+        ifile = open(self.params['custom']['custom_filename'], 'r')
+
```
Comment (mdigiorgio) [inline libs/wlgen/wlgen/rta.py]: Still missing 3 spaces in this block :smile:

Comment (credp) [inline libs/wlgen/wlgen/rta.py]: hahaha, which editor are you using? I can't persuade any of mine to show me the whitespaces.

Comment (credp) [inline libs/wlgen/wlgen/rta.py]: that's odd, there are 8 spaces before both those lines. what settings are you after?

Comment (mdigiorgio) [inline libs/wlgen/wlgen/rta.py]: Try to use vim with:

```python
:set tabstop=4
:set softtabstop=4
:set shiftwidth=4
:set textwidth=79
:set expandtab
:set autoindent
:set fileformat=unix
```

Those are my settings for `python`

**Pull request closed without merging**